File: ios/ZillitLCW/Models/ScheduleDay.swift
import Foundation

struct ScheduleDay: Identifiable, Codable {
    let id: String
    let projectId: String
    var title: String
    let typeId: String
    var typeName: String
    var color: String
    let dateRangeType: String
    var startDate: Int64
    var endDate: Int64
    var numberOfDays: Int
    var calendarDays: [Int64]
    let timezone: String
    var version: Int
    let deleted: Int64
    let createdAt: Int64
    let updatedAt: Int64
    // Nested from /calendar endpoint
    var events: [ScheduleEvent]?
    var notes: [ScheduleEvent]?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case projectId, title, typeId, typeName, color, dateRangeType
        case startDate, endDate, numberOfDays, calendarDays, timezone
        case version, deleted, createdAt, updatedAt, events, notes
    }
}

// MARK: - Mock Data
extension ScheduleDay {
    static let mockDays: [ScheduleDay] = {
        let apr12: Int64 = 1744416000000
        let oneDay: Int64 = 86400000
        return [
            ScheduleDay(
                id: "d1", projectId: "p1", title: "Beach Week", typeId: "2", typeName: "Shoot", color: "#E74C3C",
                dateRangeType: "by_days", startDate: apr12 + 2*oneDay, endDate: apr12 + 3*oneDay,
                numberOfDays: 2, calendarDays: [apr12 + 2*oneDay, apr12 + 3*oneDay],
                timezone: "UTC", version: 1, deleted: 0, createdAt: apr12, updatedAt: apr12,
                events: nil, notes: nil
            ),
            ScheduleDay(
                id: "d2", projectId: "p1", title: "", typeId: "3", typeName: "Wrap", color: "#27AE60",
                dateRangeType: "by_days", startDate: apr12, endDate: apr12 + 5*oneDay,
                numberOfDays: 6, calendarDays: [apr12, apr12+oneDay, apr12+2*oneDay, apr12+3*oneDay, apr12+4*oneDay, apr12+5*oneDay],
                timezone: "UTC", version: 1, deleted: 0, createdAt: apr12, updatedAt: apr12,
                events: nil, notes: nil
            ),
        ]
    }()
}

File: android/app/src/main/java/com/zillit/lcw/data/model/ScheduleDay.kt
package com.zillit.lcw.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ScheduleDay(
    @SerialName("_id") val id: String,
    val projectId: String,
    var title: String = "",
    val typeId: String,
    var typeName: String = "",
    var color: String = "#3498DB",
    val dateRangeType: String = "by_dates",
    var startDate: Long = 0L,
    var endDate: Long = 0L,
    var numberOfDays: Int = 0,
    var calendarDays: List<Long> = emptyList(),
    val timezone: String = "UTC",
    var version: Int = 1,
    val deleted: Long = 0L,
    val createdAt: Long = 0L,
    val updatedAt: Long = 0L,
    // Calendar endpoint includes nested events/notes
    val events: List<ScheduleEvent> = emptyList(),
    val notes: List<ScheduleEvent> = emptyList()
)
